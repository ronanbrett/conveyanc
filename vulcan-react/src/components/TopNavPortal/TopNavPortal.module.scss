@import "src/styles/vars";

.TopNavPortal {

}