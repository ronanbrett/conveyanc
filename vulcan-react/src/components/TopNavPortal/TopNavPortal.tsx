import React, { useEffect } from "react";
import { createPortal } from "react-dom";

const TopNavPortal: React.FC<any> = ({ children }: { children: any }) => {
  const topNavPortal = document.getElementById("topNavPortal") as HTMLElement;
  const el = document.createElement("div");

  useEffect(() => {
    topNavPortal.appendChild(el);
  });

  useEffect(() => {
    return () => topNavPortal.removeChild(el) as any;
  });

  return createPortal(children, el);
};

export default TopNavPortal;
