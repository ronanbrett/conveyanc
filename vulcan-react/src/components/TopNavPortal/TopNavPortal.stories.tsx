import React from 'react';

import TopNavPortal from './TopNavPortal';

import { action } from '@storybook/addon-actions';

export default {
  title: 'Components/TopNavPortal',
  decorators: [],
  component: TopNavPortal,
};


export const Default = () => (
  <TopNavPortal></TopNavPortal>
);
