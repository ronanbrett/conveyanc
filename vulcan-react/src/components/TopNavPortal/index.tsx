import TopNavPortal from './TopNavPortal';

export default TopNavPortal;