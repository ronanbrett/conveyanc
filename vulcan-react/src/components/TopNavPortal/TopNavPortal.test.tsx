import React from 'react';
import { cleanup, render } from '@testing-library/react';
import TopNavPortal from './TopNavPortal';


const generatePortal =  () => {
   let modalRoot = document.createElement("div");
   modalRoot.setAttribute("id", "topNavPortal");
   document.querySelector("body")!.appendChild(modalRoot);
}


describe('TopNavPortal', () => {

   beforeEach(generatePortal);

   afterEach(cleanup);

   it('renders without error', () => {
      const { asFragment } = render(<TopNavPortal></TopNavPortal>);
      expect(asFragment()).toMatchSnapshot();
   });
});