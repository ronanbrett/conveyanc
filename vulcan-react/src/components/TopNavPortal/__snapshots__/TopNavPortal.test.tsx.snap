// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopNavPortal renders without error 1`] = `<DocumentFragment />`;
